dbt/include/bigquery/macros/materializations/materialized_view/materialized_view.sql

```diff
@@ -0,0 +1,38 @@
+{% materialization materialized_view, adapter='bigquery' %}
+
+    {% set existing_relation = load_relation(this) %}
+    {% set target_relation = this.incorporate(type=this.MaterializedView) %}
+    {% set temp_relation = make_temp_relation(target_relation) %}
+
+    {{ run_hooks(pre_hooks, inside_transaction=False) }}
+
+        {% set build_sql = materialized_view_get_build_sql(existing_relation, target_relation, temp_relation) %}
+
+        {% if build_sql == '' %}
+            {{ materialized_view_execute_no_op(target_relation) }}
+        {% else %}
+            {{ materialized_view_execute_build_sql(build_sql, existing_relation, target_relation, post_hooks) }}
+        {% endif %}
+
+    {{ run_hooks(post_hooks, inside_transaction=False) }}
+
+    {{ return({'relations': [target_relation]}) }}
+
+{% endmaterialization %}
+
+
+{% materialized_view_get_build_sql(existing_relation, target_relation, backup_relation, intermediate_relation) %}
```

Comment:
I also think this could be replaced by choosing appropriate macros in `macros/relations/materialized_view/*.sql`. For example, the alter macro should just call replace. Also, since we're returning an empty changeset, we shouldn't get past the call to refresh.